src/core/src/op/constant.cpp

```diff
@@ -625,6 +625,10 @@ bool Constant::constant_fold(OutputVector&, const OutputVector&) {
     return false;
 }
 
+bool Constant::can_constant_fold(const OutputVector& input_values) {
```

Comment:
For nodes like this where `constant_fold` returns always false, override `can_constant_fold` will be sufficient because base class will return false also. There will less overridden functions.